### ISSUE TITLE
work with partial annotation

### ISSUE THREAD

**Issue by dalessioluca** [spacetx/spacetx-research#8]
If few images are partially annotated then you can do supervised learning. 
I would think that given an integer_mask_annotation:

1. compute bounding box, centroid and width/height (using skimage)
2. Put a direct regression loss with tx_map, ty_map, tw_map, th_map with are all in (0,1). 
3. All location inside the bounding box should have a lossi between p_map and target probability. The target probability is 1 at the center of the bounding box and zero in all the other location of the bounding box, i.e. the probability is both pushed up (at center) and down (at periphery) 
3. identifies the bb with the most IoU with the target bb. For that bb put a classification loss between the inferred and target mask
4. if there is no annotation this additional loss should be zero.

**Comment by dalessioluca** [comment]
Asking the model to segment object starting from images is a crazy request.
It should not be possible. 
It it only possible if:
1. movie
2. same scene from different point of view 
By the way, predicting one z-slice from a different z-slice might be a good a approach

**Comment by dalessioluca** [comment]
The data loader would need to be changed to allow for both cases when annotation is present/not present

**Comment by dalessioluca** [comment]
Another possibility is to have:

1. aggressive pre-porcessing
2. pre-training based on disks (see line 380 in https://github.com/spacetx/spacetx-research/blob/38_OMG/MODULES/vae_parts.py)

When a model is pretrained we should report the mean and std of each channel in the dataset used for training. User will be responsible to match the mean and std in order to get the most out of the pretrained model.
